lib/roo_on_rails/context_logging.rb

```diff
@@ -0,0 +1,80 @@
+require 'active_support/core_ext/module/delegation'
+require 'active_support/core_ext/object/blank'
+require 'logger'
+require 'active_support/logger'
+
+module RooOnRails
+  # Wraps any standard logger to provide context, similar to `ActiveSupport::TaggedLogging`
+  # but with key/value pairs that are appended to the end of the text.
+  #
+  #   logger = RooOnRails::ContextLogging.new(Logger.new($stdout))
+  #   logger.context(a: 1, b: 2) { logger.info 'foo' }                     # Logs "foo -- a=1 b=2"
+  #   logger.context(a: 1) { logger.context(b: 2) { logger.info('foo') } } # Logs "foo -- a=1 b=2"
+  #
+  # You can also use `push_context` and `pop_context` in before/after blocks.
+  #
+  #   class ApplicationController < ActionController::Base
+  #     before_action { logger.push_context(user_id: current_user.id) }
+  #     after_action { logger.pop_context }
+  #   end
+  module ContextLogging
+    module Formatter
+      def call(severity, timestamp, progname, msg)
+        super(severity, timestamp, progname, "#{msg}#{context_text}")
+      end
+
+      def context(**context)
+        push_context(**context)
+        yield self
+      ensure
+        pop_context
+      end
+
+      def push_context(**context)
```

Comment:
Nothing really... I was just copying what `ActiveSupport::TaggedLogging` has. It makes some things like before/after action blocks slightly cleaner as you don't have to worry about error handling in an around_action block. But happy to remove as it clearly _is_ running with scissors.